tests/pipelines/test_pipelines_common.py

```diff
@@ -59,6 +61,105 @@ def check_same_shape(tensor_list):
     return all(shape == shapes[0] for shape in shapes[1:])
 
 
+class IPAdapterTesterMixin:
+    """
+    This mixin is designed to be used with PipelineTesterMixin and unittest.TestCase classes.
+    It provides a set of common tests for pipelines that support IP Adapters.
+    """
+
+    def test_pipeline_signature(self):
+        parameters = inspect.signature(self.pipeline_class.__call__).parameters
+
+        assert issubclass(self.pipeline_class, IPAdapterMixin)
+        self.assertIn(
+            "ip_adapter_image",
+            parameters,
+            "`ip_adapter_image` argument must be supported by the `__call__` method",
+        )
+        self.assertIn(
+            "ip_adapter_image_embeds",
+            parameters,
+            "`ip_adapter_image_embeds` argument must be supported by the `__call__` method",
+        )
+
+    def _get_dummy_image_embeds(self, cross_attention_dim: int = 32):
+        return torch.randn((2, 1, cross_attention_dim), device=torch_device)
+
+    def _modify_inputs_for_ip_adapter_test(self, inputs: Dict[str, Any]):
+        inputs["output_type"] = "np"
+        inputs["return_dict"] = False
+        if "image" in inputs.keys():
+            inputs["num_inference_steps"] = 4
```

Comment:
Just a question. Why increase this based on the whether `image` is provided? Because of `strength`?

Reply:
That is correct. The tests fail for img2img tests that have num_inference_steps set to 2 or lower and strength greater than 5. I think if it was:
```py
if "image" in inputs().keys() and "strength" in inputs().keys():
    ...
```
it would make more sense.